src/function/vector_string_functions.cpp

```diff
@@ -23,74 +23,69 @@ using namespace kuzu::common;
     uint32_t resultLen = getResultLen((char*)input.getData(), input.len, isUpper);
     result.len = resultLen;
-    if (resultLen <= common::ku_string_t::SHORT_STR_LENGTH) {
+    if (resultLen <= ku_string_t::SHORT_STR_LENGTH) {
         convertCase((char*)result.prefix, input.len, (char*)input.getData(), isUpper);
     } else {
         result.overflowPtr = reinterpret_cast<uint64_t>(
-            common::StringVector::getInMemOverflowBuffer(&resultValueVector)
-                ->allocateSpace(result.len));
+            StringVector::getInMemOverflowBuffer(&resultValueVector)->allocateSpace(result.len));
         auto buffer = reinterpret_cast<char*>(result.overflowPtr);
         convertCase(buffer, input.len, (char*)input.getData(), isUpper);
-        memcpy(result.prefix, buffer, common::ku_string_t::PREFIX_LENGTH);
+        memcpy(result.prefix, buffer, ku_string_t::PREFIX_LENGTH);
     }
 }
 
-void BaseStrOperation::operation(common::ku_string_t& input, common::ku_string_t& result,
-    common::ValueVector& resultValueVector, uint32_t (*strOperation)(char* data, uint32_t len)) {
-    if (input.len <= common::ku_string_t::SHORT_STR_LENGTH) {
+void BaseStrOperation::operation(ku_string_t& input, ku_string_t& result,
+    ValueVector& resultValueVector, uint32_t (*strOperation)(char* data, uint32_t len)) {
+    if (input.len <= ku_string_t::SHORT_STR_LENGTH) {
         memcpy(result.prefix, input.prefix, input.len);
         result.len = strOperation((char*)result.prefix, input.len);
     } else {
         result.overflowPtr = reinterpret_cast<uint64_t>(
-            common::StringVector::getInMemOverflowBuffer(&resultValueVector)
-                ->allocateSpace(input.len));
+            StringVector::getInMemOverflowBuffer(&resultValueVector)->allocateSpace(input.len));
         auto buffer = reinterpret_cast<char*>(result.overflowPtr);
         memcpy(buffer, input.getData(), input.len);
         result.len = strOperation(buffer, input.len);
         memcpy(result.prefix, buffer,
-            result.len < common::ku_string_t::PREFIX_LENGTH ? result.len :
-                                                              common::ku_string_t::PREFIX_LENGTH);
+            result.len < ku_string_t::PREFIX_LENGTH ? result.len : ku_string_t::PREFIX_LENGTH);
     }
 }
 
 void Concat::concat(const char* left, uint32_t leftLen, const char* right, uint32_t rightLen,
-    common::ku_string_t& result, common::ValueVector& resultValueVector) {
+    ku_string_t& result, ValueVector& resultValueVector) {
     auto len = leftLen + rightLen;
-    if (len <= common::ku_string_t::SHORT_STR_LENGTH /* concat result is short */) {
+    if (len <= ku_string_t::SHORT_STR_LENGTH /* concat result is short */) {
         memcpy(result.prefix, left, leftLen);
         memcpy(result.prefix + leftLen, right, rightLen);
     } else {
         result.overflowPtr = reinterpret_cast<uint64_t>(
```

Comment:
`static ku_string_t& reserveString(ValueVector* vector, uint32_t vectorPos, uint64_t length);`
You can utilize reserveString in the StringVector.